### PR TITLE
Feature/ms 8852 proper software name in kmehr

### PR DESCRIPTION
**Pull request by sam-medispring**

### CODE REVIEW

File: src/main/kotlin/org/taktik/icure/be/ehealth/logic/kmehr/incapacity/impl/v20170601/IncapacityExport.kt [hunk 1]
```diff
@@ -355,6 +360,12 @@ class IncapacityExport(
 									contents.add(
 										ContentType().apply {
 											person = PersonType().apply {
+												//NOTE: Mult-eMediatt only requires the phone number of the contactperson and no other data.
+												//		But kmehr requires name/id/sex so dummy values are given here
+												ids.add(IDPATIENT().apply { s = IDPATIENTschemes.LOCAL; sv = config.soft?.version; sl = "${config.soft?.name}-Person-Id"; value = "cp" })
+												firstnames.add("contactpersontel")
+												familyname = "contactpersontel"
```
Comment (aduchate) [inline src/main/kotlin/org/taktik/icure/be/ehealth/logic/kmehr/incapacity/impl/v20170601/IncapacityExport.kt:367]: contactpersontel ?

**Review by aduchate — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/taktik/icure/be/ehealth/logic/kmehr/incapacity/impl/v20170601/IncapacityExport.kt (hunk 1 above)
Comment (sam-medispring) [reply src/main/kotlin/org/taktik/icure/be/ehealth/logic/kmehr/incapacity/impl/v20170601/IncapacityExport.kt:367]: For mult-emediatt we only have the telephone number, but kmehr persontype requires names/id/sex so we provide dummy values here, it's uggly but it's what ehealth wants. (do you want me to put the more neutral 'dummyName' ?)

File: src/main/kotlin/org/taktik/icure/domain/be/kmehr/IncapacityExportInfo.kt [hunk 2]
```diff
@@ -39,4 +39,6 @@ data class IncapacityExportInfo(
 	val recoveryAddress: Address? = null,
 	val foreignStayBegin: Long = 0, //MMEDIATT-ITEM foreignstay
 	val foreignStayEnd: Long = 0,
+	val softwareName: String? = null,
```
Comment (aduchate) [inline src/main/kotlin/org/taktik/icure/domain/be/kmehr/IncapacityExportInfo.kt:42]: It is weird to set it at "" when you use null to fallback to the default values. That means that the default values will never be used.

File: src/main/kotlin/org/taktik/icure/services/external/rest/v1/dto/be/kmehr/IncapacityExportInfoDto.kt [hunk 3]
```diff
@@ -43,4 +43,6 @@ data class IncapacityExportInfoDto(
 	val recoveryAddress: AddressDto? = null,
 	val foreignStayBegin: Long = 0, //MMEDIATT-ITEM foreignstay
 	val foreignStayEnd: Long = 0,
+	val softwareName: String? = "",
+	val softwareVersion: String? = ""
```
Comment (aduchate) [inline src/main/kotlin/org/taktik/icure/services/external/rest/v1/dto/be/kmehr/IncapacityExportInfoDto.kt:47]: Same

File: src/main/kotlin/org/taktik/icure/services/external/rest/v2/dto/be/kmehr/IncapacityExportInfoDto.kt [hunk 4]
```diff
@@ -37,4 +37,6 @@ class IncapacityExportInfoDto : Serializable {
 	var recoveryAddress: AddressDto? = null
 	var foreignStayBegin: Long = 0 //MMEDIATT-ITEM foreignstay
 	var foreignStayEnd: Long = 0
+	var softwareName: String? = ""
+	var softwareVersion: String? = ""
```
Comment (aduchate) [inline src/main/kotlin/org/taktik/icure/services/external/rest/v2/dto/be/kmehr/IncapacityExportInfoDto.kt:41]: Same

**Review by aduchate — CHANGES_REQUESTED** [review]

**Pull request merged**